### ISSUE TITLE
Sinks with explicit consistency formats do not work with reuse_topic

### ISSUE THREAD

**Issue by elindsey** [MaterializeInc/materialize#8231]
Currently we have a parser bug that prevents users from creating JSON formatted sinks with Avro consistency topics and reuse_topic=true

The expected syntax that should work is:
```
> CREATE SINK json_reuse_topic FROM unnamed_cols                                                                                                                                                                                                                            
  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'json-reuse-topic'
    CONSISTENCY TOPIC 'consistency-json-avro' CONSISTENCY FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
    WITH (reuse_topic=true)
  FORMAT JSON
```

However, the WITH clause is slurped up as part of the schema registry parsing, meaning the `reuse_topic` is ignored and this will create a sink with a nonced name.

Today you can trick the parser by doing this:
```
> CREATE SINK json_reuse_topic FROM unnamed_cols                                                                                                                                                                                                                            
  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'json-reuse-topic'
    CONSISTENCY TOPIC 'consistency-json-avro' CONSISTENCY FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
    WITH (_=_)
    WITH (reuse_topic=true)
  FORMAT JSON
```

Q: Why wasn't this caught by unit tests?
A: The unit tests are using testdrive's `kafka-verify`, which does a lookup in the catalog to find the exact topic name - meaning it was verifying results using the nonced topic name. The tests themselves are also incorrect, they attempt to use a view as input to an exactly-once sink, which is unsupported. Once the parser issue if fixed, new tests need to be written and we need to get rid of this testdrive footgun.

Q: Does this affect avro sinks with reuse_topic?
A: If an avro sinks is created with the consistency topic explicitly specified as FORMAT AVRO, then yes. This is not common as the default consistency topic format is avro.

Most of the relevant code for this is in parser.rs, the call tree that follows from `parse_create_sink`.

**Comment by aljoscha** [comment]
Note: we could add a parser test [here](https://github.com/MaterializeInc/materialize/blob/3792391c14f09701bdca9052f65160e2dba0e128/src/sql-parser/tests/testdata/ddl#L667) in addition to the other things Eli mentioned.

**Comment by JLDLaughlin** [comment]
So, this is sort of fixed! Per #8370, we will continue to parse the buggy syntax. However, we now support a new, non-buggy syntax that is [officially outlined in our docs](https://materialize.com/docs/sql/create-sink/#sink_kafka_connector).

I'm going to close this in favor of https://github.com/MaterializeInc/materialize/issues/8411.
